### DISCUSSION TITLE
Are function arguments condtants or variable ?

### DISCUSSION THREAD

**Discussion by sanjayjain159** [janet-lang/janet#1428] [General]
Typically in a programming language, the function arguments are local variables. But as far as a noobie like me knows Janet treats function arguments a b c declared as in, (defn f [a b c] ...) as constants. How to get them as local variables whose value can be changed for example with set?

**Comment by bakpakin** [comment]
You cannot do that. I believe you are referring to "inout" parameters from other languages?

**Comment by avitkauskas** [comment]
But what about passing a mutable variable as parameter?
I tried this in REPL:

```
(def tab @{})
(defn fun [t] (put t :k "v"))
(fun tab)
(pp tab)
=> @{:k "v"}
```

Feels like "inout"..

**Reply by bakpakin** [reply]
Yes, this is the only way to do this. Function parameters are equivalent to bindings made with 'def'.
